sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -158,6 +158,9 @@ trait CheckAnalysis extends PredicateHelper {
           case g: GroupingID =>
             failAnalysis("grouping_id() can only be used with GroupingSets/Cube/Rollup")
 
+          case Alias(w: WindowFunction, _) =>
```

Comment:
BTW why do we have to match the `Alias`?

Reply:
The right way is `Alias(WindowExpression(w: WindowFunction, _),  _)` and without `over` it is `Alias(w: WindowFunction, _)`.

Reply:
then the check doesn't work for cases like `rank() + 1`?

Reply:
good catch ! we need to check all expression.